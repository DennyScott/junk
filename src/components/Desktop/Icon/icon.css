.app-name{
    font-size: 12px;
    color: white;
    font-weight: bolder;
}

.desktop-icon {
    cursor: pointer;
}