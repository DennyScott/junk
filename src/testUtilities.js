import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

export const rendersCorrectly =  component => {
  const wrapper = shallow(component);
  expect(wrapper.length).toBe(1);
}

export const matchesSnapshot = component => {
  const wrapper = shallow(component);
  expect(toJson(wrapper)).toMatchSnapshot();
}