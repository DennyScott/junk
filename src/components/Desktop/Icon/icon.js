import React, { Component } from 'react';
import './icon.css';
import Draggable from 'react-draggable';

export default class Icon extends Component {
  render() {
    return (
      <Draggable handle=".handle">
          <div className="desktop-icon handle">
            <img src={this.props.img} width="32" height="32" draggable="false"/>
          </div>
        </Draggable>
    )
  }
}
