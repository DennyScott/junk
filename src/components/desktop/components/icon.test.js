import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { rendersCorrectly, matchesSnapshot } from 'testUtilities';

import Icon from './icon';

const minProps = {
  name: '',
  logo: '',
  onDoubleClick: jest.fn(),
};

const getMinComponent = () => <Icon {...minProps} />;

it('Desktop renders correctly', () => {
  rendersCorrectly(getMinComponent(), expect);
});

it('Desktop matches snapshot correctly', () => {
  matchesSnapshot(getMinComponent(), expect);
});

it('Double clicking the icon calls the double click prop', () => {
  const wrapper = shallow(getMinComponent());
  const spy = jest.spyOn(minProps, 'onDoubleClick');
  wrapper.find('.desktop-icon').simulate('doubleClick');
  expect(spy).toHaveBeenCalled();
});

it('The app name is displayed', () => {
  const appName = 'appName';
  const wrapper = shallow(<Icon {...minProps} name={appName} />);
  expect(toJson(wrapper.find('.app-name').children())).toBe(appName);
});
