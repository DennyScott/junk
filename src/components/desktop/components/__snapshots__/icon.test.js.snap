// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop matches snapshot correctly 1`] = `
<Draggable
  allowAnyClick={false}
  axis="both"
  bounds={false}
  cancel={null}
  defaultClassName="react-draggable"
  defaultClassNameDragged="react-draggable-dragged"
  defaultClassNameDragging="react-draggable-dragging"
  defaultPosition={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  disabled={false}
  enableUserSelectHack={true}
  grid={null}
  handle=".handle"
  offsetParent={null}
  onDrag={[Function]}
  onMouseDown={[Function]}
  onStart={[Function]}
  onStop={[Function]}
  position={null}
  transform={null}
>
  <styled.div
    className="desktop-icon handle"
    onDoubleClick={[Function]}
  >
    <img
      alt="logo"
      draggable="false"
      height="42"
      src=""
      width="42"
    />
    <styled.div>
      <span
        className="app-name"
      />
    </styled.div>
  </styled.div>
</Draggable>
`;
